dirtify.py

```diff
@@ -60,41 +62,22 @@
 if '.DS_Store' in images:
     images.remove('.DS_Store') # For macOS: Skip invisible Desktop Services Store file.
 
-for i, img in tqdm(enumerate(images), total=len(images)):
-    filename = f"{raw_dir}/{images[i]}"
-    img = cv2.imread(filename, cv2.IMREAD_COLOR)
-    
-    # resize and crop to make a 64 x 64 image
-    in_h = img.shape[0]
-    in_w = img.shape[1]
-    scale = max(out_h / in_h, out_w / in_w)
-
-    # resize...
-    img = cv2.resize(img, None, fx = scale, fy = scale, interpolation = cv2.INTER_AREA)
-    mid_h = scale * in_h / 2
-    mid_w = scale * in_w / 2
-
-    # ...and crop
-    img = img[round(mid_h - out_h/2):round(mid_h + out_h/2), round(mid_w - out_w/2):round(mid_w + out_w/2)]
-    out_name = os.path.splitext(images[i])[0] + ".png" # Change the extension to png (for lossless image)
-    cv2.imwrite(f"{clean_dir}/{out_name}", img)
-
-    # Add that dirt!
-    dirty = img
-    
+# Dirtify function
+def add_dirt(in_img, w, h):
+
     if args.invert:
         # Invert the image
-        dirty = 255 - dirty
+        img = 255 - in_img
```

Comment:
We need to use `in_img` throughout this function. Each of these `if` blocks is making its own copy of the image and assigning it to the variable `img`. That means only the last one that we run through is going to take effect.

Reply:
Oh yeah, that's true. Good catch; I only tested with single filters, not combinations.